enhancements/baremetal/an-slo-for-baremetal.md

```diff
@@ -0,0 +1,308 @@
+
+## Summary
+
+The Bare Metal Operator provides bare metal machine management
+capabilities needed for the Machine API provider on the `baremetal`
+platform, and there is no equivalent for this component on other
+platforms.
+
+The Bare Metal Operator is currently managed by the Machine API
+Operator in a fashion that requires the Machine API Operator to have
+significant bare metal specific knowledge. This proposal outlines a
+plan for the Bare Metal Operator to become a fully-fledged Second
+Level Operator under the management of the Cluster Version Operator.
+
+(To avoid likely confusion, please note that this proposal describes a
+new project tentatively named Bare Metal Operator (BMO), implying that
+the current baremetal-operator project will be renamed to reflect the
+fact it is "just a controller").
+
+## Motivation
+
+In order to bring up a cluster using the baremetal platform:
+
+1. The installer needs to capture bare metal host information and
+   provisioning configuration for later use.
+2. Something needs to install the CRDs for these bare metal specific
+   resources created by the installer.
+3. Something needs to launch a controller for these resources.
+
+Currently (1) is achieved by the installer creating:
+
+* A Provisioning resource manifest
+* Manifests for BareMetalHost resources and their associated secrets
+
+and these manifests are applied by the cluster-bootstrap component
+towards the end of the cluster bootstrapping process.
+
+This resource creation step does not succeed until the step (2)
+completes - i.e. the relevant CRDs are applied - and this is currently
+done by the Cluster Version Operator (CVO) as it applies the manifests
+for the MAO.
+
+Finally, (3) happens when the MAO detects that it is running on the
+`baremetal` infrastructure platform and instantiates a relatively
+complex Deployment including the BareMetalHost controller and various
+containers running other components, all from the Metal3 project. The
+configuration of this deployment is driven by the Provisioning
+resource created by the installer in (1).
+
+There are two problems emerging with this design:
+
+* A sense that the MAO is responsible for a significant amount of bare
+  metal specific tasks that should be declared outside of the scope of
+  the MAO, particularly since it does not have equivalent
+  responsibilities on any other platform.
+
+* Expanding needs for bare metal specific manifests - for example, new
+  CRDs used to drive new BMO capabilities - to be installed early in
+  the cluster bring-up means introducing yet more bare metal specific
+  concerns into the MAO.
+
+Steps (2) and (3) are aspects of cluster bring-up which the CVO is
+clearly well-suited. However, to date, it was understood that creating
+a Second Level Operator (SLO) (in other words, a CVO-managed operator)
+for bare metal would not make sense, since it implied a component
+installed and running on clusters where it is not needed.
```

Comment:
so it does mean that on any other cluster that is not bm, the BM operator will run, correct? Or there and API to enable/disable SLOs?